Driveable Death Explosion Changes

conman180 opened the pull request:
I copied some code from ShootableType.java and EntityGrenade.java over to DriveableType and EntiyDriveable to give control over the explosion if IsExplosionWhenDestroyed is set to true.

This would allow crashing a vehicle to be destructive rather than just making blocks disappear when hit.
An example that could be made is Kamikaze aircraft or a car bomb.

Hopefully this can make a fun addition to the mod.

conman180 committed:
Update EntityDriveable.java

Changes Where Explosion is from, allows control over death explosion of driveables

conman180 committed:
Update DriveableType.java

Copied code from ShootableType to control explosions

conman180 commented:
Here is an example clip I made to show it works However plane physics are different here to to my personal meddling with the flight controller. TBH I had  a bit too much fun while testing this.
![PlaneShootdownWithBoom](https://user-images.githubusercontent.com/78190656/130736594-a097fb66-3de7-469f-9fa2-25d9ebab5f53.gif)

GoldSloth reviewed (commented):
Nice work, these provide a lot of customisability to the explosion. One thing I'd request changed though, is to prefix both the variable names and the config keywords to indicate that they're for configuring the death explosion, otherwise it's a bit vague imo. It'd also be great if you could add these to the config reference too, but I'm happy to do that otherwise.

GoldSloth commented:
I'll also take a look at the explosion logic later, I'm not sure both types of explosions are needed - but I'm sure it works regardless. Many thanks for your contributions!

conman180 committed:
Update DriveableType.java

conman180 committed:
Update EntityDriveable.java

conman180 commented:
I altered those commits, should be a separate set of config options for death explosions. I also added them to the config reference

GoldSloth commented on src/main/java/com/flansmod/common/driveables/EntityDriveable.java:
```diff
@@ -2471,7 +2472,36 @@ public void checkParts() {
+        	        }
+        	        else
+        	        {
+        	        	worldObj.createExplosion(this, posX, posY, posZ, type.explosionRadius, TeamsManager.explosions && type.explosionBreaksBlocks);
```
Sorry, just got a chance to look at this again. If I understand it correctly the new logic is:

```Java
if (type.IsExplosionWhenDestroyed) {
    // Flans big boom fancy explosion
} else {
    // Vanilla explosion
}
```
To keep consistent with the old way it worked, I'd  suggest removing the else, it's either a custom Flans explosion (People can remove the destructiveness if they need) or no explosion at all. Do you agree with that?

conman180 replied:
I agree. Ill remove the else

GoldSloth reviewed (changes requested):
rip, forgot to submit this review last night .-.

conman180 committed:
Update EntityDriveable.java

remove else logic from death explosion. Either it is a flansmod explosion or none at all.

GoldSloth commented:
Awesome! Thank you for your work on this feature, I'll pull them both now.
